mistralrs-core/src/engine/mod.rs

```diff
@@ -103,6 +103,8 @@ impl Engine {
         let rng = Arc::new(std::sync::Mutex::new(Isaac64Rng::seed_from_u64(SEED)));
         let mut last_completion_ids: Vec<usize> = vec![];
         'lp: loop {
+            tokio::time::sleep(std::time::Duration::from_micros(500)).await;
```

Comment:
I like the idea of not sitting in a hot loop, but I feel like always waiting 0.5ms (essentially wasting performance in the decoding stage) isn't good. Could you modify this to only happen when the scheduler has no waiting/running sequences?